doc/doxygen/src/board-cpu-features.md

```diff
@@ -0,0 +1,99 @@
+
+FEATURES                                                             {#features}
+========
+
+`FEATURES_PROVIDED` are hardware (including toolchain) features (e.g.:`periph_hwrng`,
+`periph_uart`) that are available or characteristics (e.g:`arch_32bits`) of a board.
+
+Providing a `FEATURE`
+--------------------
+
+For a `FEATURE` to be provided by a `board` it must meet 2 criteria, and for
+`periph` and other `hw` it must follow the 3rd criteria.
+
+- Needs the "hardware" or BSP support (toolchain)
+    - e.g.: `stm32l152re` has an SPI peripheral
+            `riotboot` needs to be able to link and flash at an offset
+- Needs support in RIOT, an implementation of an api to interact with the `hw`
+    - e.g.: `cpu/stm32_common/periph/spi.c` is implemented for `stm32l1`
+            `riotboot` needs an implementation of `cpu_jump_to_image`
+- Wiring between the `cpu/soc` a `bus` and other `cpu`/`hw` components.
+    - e.g.: `nucleo-l152re/include/periph_conf.h` specified wiring between `PORT_Ax`
+             and `SPI1`
+
+What is a `FEATURE`
+--------------------
+
+Whenever a `FEATURE` is used there should be at some level a HW requirements,
+whether this is a `radio`, a `bus` ot a specific core architecture. 
+
+This is not a hard line, in some cases the line can be harder to establish than
+others. There are complicated cases like `netif` since a network interface could
+be fully implemented in software as a loop-back.
+
+It's also important to note that a `FEATURE` does not mean there is a `MODULE`
+with the same name. There could be many implementations for the same `FEATURE`.
+The fact that many `FEATURES` translate directly into a `MODULE` is only by
+convenience.
+
+e.g.
+
+    # all periph features correspond to a periph submodule
+    USEMODULE += $(filter periph_%,$(FEATURES_USED))
+
+A `FEATURE` is a tool to specify valid/invalid dependencies and configurations.
```

Comment:
This could be said first. Its only goal is doing automatic detection of invalid configurations and so handle one application for all RIOT boards without explicit per board configuration step. Otherwise you just add modules and know what you are doing.

It is also currently the way to do "optional inclusion of support". Adding a short info on optional features could be good. The optional thingy is as the "default implementation" a way to configure the resolution policy more than the dependency. Include by default all what is optional.

A future step could be to never show `features` at the application level but represent them with a virtual module, and add optional module dependency as we talked with @aabadie 
It is more an implementation issue to have them there and leaks this inconsistency to the user.
"But what if my feature is provided only by software, why should it be set in the board".
It is just the current state of RIOT. If the feature can be provided only by software, then all boards having this implementation must say it. But then how to configure which implementation to choose ? and this is mostly done in a non configurable way by the magic `periph_` handling.